### PR TITLE
[docs] Update `agent_dev_env` with newest recommendations

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?

The old document here has been outdated and did not specify
some very important "gotcha"s that could end up severely
deteriorating the developer experience that are now fixed up.

### Motivation

Issues building/running the agent; AC-164

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

N/A

### CODE REVIEW

File: docs/dev/agent_dev_env.md [hunk 1]
```diff
@@ -28,11 +33,44 @@ You will also need the Visual Studio for [Visual Studio for Python installer](ht
...
+self-contained development environment and ensure a clean system Python.
+
+**Please note that due to the [way
+some virtual environments handle executable paths](https://bugs.python.org/issue22213)
+(e.g. `python -m venv`), not all vrtual environment options will be able to run the built
+agent correctly. At this time, the only confirmed virtual enviroment that is known for
+sure to work is `virtualenv`.**
+
+- Install the virtualenv module:
+`pip3 install virtualenv`
```
Comment (mx-psi) [inline docs/dev/agent_dev_env.md:55]: The recommended way of running `pip` is by using the `-m` option (see e.g. [all examples here](https://packaging.python.org/tutorials/installing-packages/))
```suggestion
`python3 -m pip install virtualenv`
```

File: docs/dev/agent_dev_env.md [hunk 2]
```diff
@@ -28,11 +33,44 @@ You will also need the Visual Studio for [Visual Studio for Python installer](ht
...
+- Create the virtual environment:
+`virtualenv $GOPATH/src/github.com/DataDog/datadog-agent/venv`
+- [Activate the virtualenv](https://virtualenv.pypa.io/en/latest/user_guide.html#activators) (OS-dependent).
+  This must be done for every new terminal before you start.
+- Specify the path when building the agent:
+`invoke agent.build --python-home-3=$GOPATH/src/github.com/DataDog/datadog-agent/venv`
+
+If you are using Python 2 instead (or switching between python versions), you can also
```
Comment (mx-psi) [inline docs/dev/agent_dev_env.md:63]:
```suggestion
If you are using Python 2 instead (or switching between Python versions), you can also
```

**Review by ogaca-dd — APPROVED** [review]
LGTM 👍 
Did you have to install `gettext` in order to build the Agent on Mac OS?

File: docs/dev/agent_dev_env.md [hunk 3]
```diff
@@ -47,39 +85,27 @@ file and `pip`:
...
-```invoke agent.build --python-home-2=$GOPATH/src/github.com/DataDog/datadog-agent/venv```
-
-If you are using python 3 instead (or switching between python versions), you can also
-add `--python-home-3=<path>` pointing to a python3 virtual environment.
-
 ## Golang
 
-You must install [go](https://golang.org/doc/install) version 1.11.5 or above. Make
-sure that `$GOPATH/bin` is in your `$PATH` otherwise Invoke cannot use any
+You must [install Golang](https://golang.org/doc/install) version `1.14`. Make
```
Comment (mx-psi) [inline docs/dev/agent_dev_env.md:100]: Can you add this as a step in the instructions for upgrading the Golang version in the mars-jenkin repo?

File: docs/dev/agent_dev_env.md [hunk 4]
```diff
@@ -47,39 +85,27 @@ file and `pip`:
...
+sure that `$GOPATH/bin` is in your `$PATH` otherwise `invoke` cannot use any
 additional tool it might need.
 
+**Please note that versions of Golang that aren't an exact match to the version
+specified in our build image [here](https://github.com/DataDog/datadog-agent/blob/master/.circleci/images/builder/Dockerfile#L1)
```
Comment (mx-psi) [inline docs/dev/agent_dev_env.md:105]: We use several build images, this one works but I think we should reword the sentence 
```suggestion
specified in our build images (see e.g. [here](https://github.com/DataDog/datadog-agent/blob/master/.circleci/images/builder/Dockerfile#L1))
```

**Review by mx-psi — COMMENTED** [review]
Thanks for doing this! I left some minor comments

File: docs/dev/agent_dev_env.md [hunk 5]
```diff
@@ -28,11 +33,44 @@ You will also need the Visual Studio for [Visual Studio for Python installer](ht
...
+sure to work is `virtualenv`.**
+
+- Install the virtualenv module:
+`pip3 install virtualenv`
+- Create the virtual environment:
+`virtualenv $GOPATH/src/github.com/DataDog/datadog-agent/venv`
+- [Activate the virtualenv](https://virtualenv.pypa.io/en/latest/user_guide.html#activators) (OS-dependent).
+  This must be done for every new terminal before you start.
+- Specify the path when building the agent:
+`invoke agent.build --python-home-3=$GOPATH/src/github.com/DataDog/datadog-agent/venv`
```
Comment (hush-hush) [inline docs/dev/agent_dev_env.md:61]: Is this needed ? It should be handle automatically when using a venv, no ?
On linux I don't need this.

If it's not automatic on OSX we should handle the case. The goal is to be able do build the agent with only `invoke agent.build`

File: docs/dev/agent_dev_env.md (hunk 3 above)
Comment (sgnn7) [reply docs/dev/agent_dev_env.md:100]: I might ping you on slack to follow-up on this

File: docs/dev/agent_dev_env.md (hunk 5 above)
Comment (sgnn7) [reply docs/dev/agent_dev_env.md:61]: As far as I could figure it out on OSX, the packages are still attempted only to be loaded from system store even if you use a virtualenv. I'll update the docs to make sure to note that this only applies to that platform.

**Comment by sgnn7** [comment]
> Did you have to install gettext in order to build the Agent on Mac OS?

@ogaca-dd I don't remember explicitly installing it but it might have been included in devenv Ansible installation.

File: docs/dev/agent_dev_env.md (hunk 3 above)
Comment (sgnn7) [reply docs/dev/agent_dev_env.md:96]: [PR created here](https://github.com/DataDog/mars-jenkins-scripts/pull/316)

**Review by mx-psi — APPROVED** [review]
Thanks for doing this!

**Review by apigirl — APPROVED** [review]
thanks for these updates!!

**Pull request merged**